microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/reactivemessaging/java/MicroProfileReactiveMessagingASTValidator.java

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4mp.jdt.internal.reactivemessaging.java;
+
+import static org.eclipse.lsp4mp.jdt.internal.reactivemessaging.MicroProfileReactiveMessagingConstants.MICRO_PROFILE_REACTIVE_MESSAGING_DIAGNOSTIC_SOURCE;
+import static org.eclipse.lsp4mp.jdt.internal.reactivemessaging.MicroProfileReactiveMessagingConstants.INCOMING_ANNOTATION;
+import static org.eclipse.lsp4mp.jdt.internal.reactivemessaging.MicroProfileReactiveMessagingConstants.OUTGOING_ANNOTATION;
+
+import java.util.List;
+import java.util.logging.Logger;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.dom.SingleMemberAnnotation;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4mp.jdt.core.java.diagnostics.JavaDiagnosticsContext;
+import org.eclipse.lsp4mp.jdt.core.java.validators.JavaASTValidator;
+import org.eclipse.lsp4mp.jdt.core.utils.JDTTypeUtils;
+
+/**
+ *
+ * MicroProfile Reactive Messaging Diagnostics:
+ *
+ * <ul>
+ * <li>Diagnostic: display reactive messaging diagnostic message if the name of
+ * the consumed channel is blank.</li>
+ * </ul>
+ *
+ * <p>
+ * This rule comes from
+ * https://github.com/eclipse/microprofile-reactive-messaging/blob/62c9ed5dffe01125941bb185f1433d6307b83c86/api/src/main/java/org/eclipse/microprofile/reactive/messaging/Incoming.java#L95
+ * </p>
+ * 
+ * @See https://github.com/eclipse/microprofile-reactive-messaging
+ *
+ */
+public class MicroProfileReactiveMessagingASTValidator extends JavaASTValidator {
+
+    private static final String BLANK_CHANNEL_NAME_MESSAGE = "The name of the consumed channel must not be blank.";
+
+    private static final Logger LOGGER = Logger.getLogger(MicroProfileReactiveMessagingASTValidator.class.getName());
+
+    @Override
+    public void initialize(JavaDiagnosticsContext context, List<Diagnostic> diagnostics) {
+        super.initialize(context, diagnostics);
+    }

```

Comment:
If you're not doing anything special with the class, you don't have to declare and have it just call the base class. You can simply leave it out. Because you are inheriting from `JavaASTValidator`, anything that calls `initialize(..)` on your class here, will just get the base anyways.